plugins/hls-refactor-plugin/src/Development/IDE/Plugin/CodeAction.hs

```diff
@@ -1087,8 +1087,9 @@ suggestImportDisambiguation df (Just txt) ps fileContents diag@Diagnostic {..}
                 ++ [HideOthers restImports | not (null restImports)]
             ] ++ [ ( renderUniquify mode T.empty symbol True
               , disambiguateSymbol ps fileContents diag symbol mode
-              ) | local, not (null targetsWithRestImports)
-                , let mode = HideOthers (uncurry (:) (head targetsWithRestImports))
+              ) | local
+                , (m,ms) <- take 1 targetsWithRestImports
+                , let mode = HideOthers (m:ms)
```

Comment:
Before this change, the plugin would crash because some precondition was violated. With this change, it will be ignored. I don't think that's strictly an improvement, I would prefer it if we  refactored this code to use `NonEmpty` or some other proof why this will always give us at least one result.

Reply:
I don't think the original impl. would crash, because it had `not (null targetsWithRestImports)` so the list comprehension would also "fail silently" returning empty list before evaluating head.
But I agree the `take 1` is confusing, so I switched to using direct pattern match on the list.